hack/update-docs.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Tekton Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source $(dirname $0)/../vendor/github.com/tektoncd/plumbing/scripts/library.sh
+cd ${REPO_ROOT_DIR}
+
+# Whether this script is being used for validation or not
+DIFF=
+# Syntax format for code fences
+SYNTAX="YAML"
+# All code fences encountered will be ignored
+FENCE_ON='^```'${SYNTAX}'$'
+FENCE_OFF='^```$'
+# Comment that will specify the file to embed within $SYNTAX code fence
+COMMENT='<!-- *FILE: *(.*) *-->'
+
+# Parse flags
+while getopts ":d" opt; do
+  case ${opt} in
+    d )
+        DIFF="true"
+        ;;
+    * )
+        echo "Invalid Option: -$OPTARG" 1>&2
+        exit 1
+        ;;
+  esac
+done
+shift $((OPTIND -1))
+
+doc_files="$(find docs -name "*.md")"
+for file in ${doc_files};do
+    new_file="${file%/*}/updated-${file##*/}"
+    > ${new_file}
+    fenced="false"
+    # Read each markdown file for replacements
+    while IFS= read -r line; do
+        if [[ ${line} =~ ${FENCE_ON} ]];then
+            fenced="true"
+            continue
+        fi
+        if [[ ${line} =~ ${FENCE_OFF} ]];then
+            fenced="false"
+            continue
+        fi
+        # Write to replacement file
+        if [[ ${fenced} == false ]];then 
```

Comment:
I think this will delete any code fences that do not have the matching comments with a file, because the line will not be copied over when `fenced=true`

Reply:
This was the way I designed it so it will always recreate the code fence when the comment is detected and then ignore the one that was already there.

Reply:
I see what you mean now. Pushed a fix.